### DISCUSSION TITLE
Creating Initial Database

### DISCUSSION THREAD

**Discussion by MGrgov90** [ravendb/ravendb#15220] [Q&A]
I'm running raven in a docker. Is there a possibility to create initial database without accessing the host. What I'm trying to do is to start docker image and fresh database on  it. 
Here is the project I'm looking into: https://github.com/testcontainers/testcontainers-dotnet .I would like to add another provider there. But I need somehow to create a database after starting an instance

**Comment by georgiosd** [comment]
You.can embed a script that does it with curl and an environment variable?

On Sat, 5 Nov 2022 at 03:53, Milos Grgov ***@***.***> wrote:

> I'm running raven in a docker. Is there a possibility to create initial
> database without accessing the host. What I'm trying to do is to start
> docker image and fresh database on it.
> Here is the project I'm looking into:
> https://github.com/testcontainers/testcontainers-dotnet .I would like to
> add another provider there. But I need somehow to create a database after
> starting an instance
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ravendb/ravendb/discussions/15220>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMYG5Q5EKSXZYNRTAA7FM3WGW4ZNANCNFSM6AAAAAARXWCZKM>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Georgios Diamantopoulos
Founder
Zero to MVP
https://zerotomvp.dev

**Comment by ayende** [comment]
Why do you need that?
In tests, most of the time you'll use a _separate_ db per test.

At any rate, you can add a startup script that would use a REST call to create the DB
